ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -367,6 +416,58 @@ private boolean updateBlobMetadata(BlobId blobId, String fieldName, Object value
     }
   }
 
+  @Override
+  public boolean purgeBlob(CloudBlobMetadata blobMetadata) throws CloudStorageException {
+    String blobId = blobMetadata.getId();
+    String blobFileName = blobMetadata.getCloudBlobName();
+    String partitionPath = blobMetadata.getPartitionId();
+    String containerName = getAzureContainerName(partitionPath);
+    RequestOptions options = new RequestOptions();
+    options.setPartitionKey(new PartitionKey(partitionPath));
+    azureMetrics.blobDeleteRequestCount.inc();
+    Timer.Context deleteTimer = azureMetrics.blobDeletionTime.time();
+    try {
+      // delete blob from storage
+      CloudBlobContainer azureContainer = azureBlobClient.getContainerReference(containerName);
+      CloudBlockBlob azureBlob = azureContainer.getBlockBlobReference(blobFileName);
+      boolean deletedSomething = azureBlob.deleteIfExists();
```

Comment:
Name it `deleteResult`?

Reply:
I think deletedSomething is more specific in this case.  I can rename to deletionDone.